third_party/outline-tun2socks/index.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {resolve, sep} from 'node:path';
+import {platform} from 'node:process';
+import {promisify} from 'node:util';
+
+import ffi from 'ffi-napi';
+import ref from 'ref-napi';
+import init_struct, {StructObject} from 'ref-struct-di';
+
+const ffiStruct = init_struct(ref);
+
+// native function definitions
+let nativeLib : {
+  readonly CheckConnectivity: ffi.ForeignFunction<StructObject<{
+    status: number,
+    pErr: ref.Pointer<unknown>,
+  }>, []>,
+
+  readonly ReleaseObject: ffi.ForeignFunction<void, [ref.Pointer<unknown>]>,
+} | undefined;
+
+
+const NATIVE_LIB_PATH_BY_PLATFORM = new Map<string, string[]>([
+  ['linux', ['prebuilds', 'linux-x64', 'libtun2socks.so']],
```

Comment:
For the prebuilt binaries, it would be better if we could fetch from the release, otherwise we will get complaints about the binaries in the source code.
Perhaps we can fetch on demand at build time.

Reply:
Good point. Also once we publish it to npm it won't cause any problems because it is downloaded to the node_modules folder.